syntax = "proto3";

package minitwitter;

service MiniTwitter {
  rpc SendMessage (SendMessageRequest) returns (SendMessageResponse);
  rpc GetMessages (GetMessagesRequest) returns (GetMessagesResponse);
}

message SendMessageRequest {
  string content = 1;
}

message SendMessageResponse {
  bool success = 1;
}

message GetMessagesRequest {
  int32 n = 1;
}

message GetMessagesResponse {
  repeated Message messages = 1;
}

message Message {
  string content = 1;
  int64 timestamp = 2;
}